test/healthcheck/framework/healthcheck.go

```diff
@@ -0,0 +1,382 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"errors"
+	goflag "flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	v1beta1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	util "github.com/kubernetes-incubator/service-catalog/test/util"
+	"github.com/spf13/cobra"
+	pflag "github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var options *HealthCheckServer
+
+func Execute() error {
+	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+	options = NewHealthCheckServer()
+	options.AddFlags(pflag.CommandLine)
+	defer glog.Flush()
+	return rootCmd.Execute()
+
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "healthcheck",
+	Short: "healtchcheck performs an end to end verification of Service Catalog",
+	Long:  `Runs a quick end to end health check for Service Catalog`,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := initialize(options)
+		if err != nil {
+			glog.Errorf("Error initialzing: %v", err)
+			os.Exit(1)
+		}
+
+		// Start the HTTP server that enables us to serve /healtz and /metrics.   The  metrics can be pulled,
+		// analyzed and alerted on.
+		err = ServeHttp(options)
+		if err != nil {
+			glog.Errorf("Error starting HTTP: %v", err)
+			os.Exit(1)
+		}
+
+		glog.Infof("Sheduled health checks will be run every %v", options.HealthCheckInterval)
+
+		// Every X interval run the health check
+		ticker := time.NewTicker(options.HealthCheckInterval)
+		for range ticker.C {
+			err = healthCheck(options)
+			if err != nil {
+				glog.Errorf("Error running health check: %v", err)
+				cleanup()
+				ErrorCount.WithLabelValues(err.Error()).Inc()
+			} else {
+				glog.V(2).Info("Successfully ran health check")
+			}
+		}
+	},
+}
+
+var (
+	// A Kubernetes and Service Catalog client
+	kubeClientSet           kubernetes.Interface
+	serviceCatalogClientSet clientset.Interface
+
+	// Namespace in which all test resources should reside
```

Comment:
So, this assumes that the ups broker is deployed, right?

I wonder if we ought to make these things CLI params.

Reply:
@pmorie, yes, as listed in the description, `At present it is hard coded to use the User Provided Broker and its associated services and plans but this will be parameterized.`